hil-test/tests/spi_half_duplex_read.rs

```diff
@@ -101,4 +101,66 @@ mod tests {
 
         assert_eq!(dma_rx_buf.as_slice(), &[0xFF; DMA_BUFFER_SIZE]);
     }
+
+    #[test]
+    #[timeout(3)]
+    fn test_spidmabus_reads_correctly_from_gpio_pin() {
```

Comment:
Would it be an unreasonable ask from me to rebase this PR on top of https://github.com/esp-rs/esp-hal/pull/1990 with the associated cleanups?